neqo-common/src/codec.rs

```diff
@@ -470,15 +457,17 @@ mod tests {
 
     #[test]
     fn decode() {
-        let enc = Encoder::from_hex("012345");
+        let mut out = vec![];
+        let enc = Encoder::new(&mut out).from_hex("012345");
```

Comment:
Might be able to make the diff shorter by doing something like
```
let enc = Encoder::new(&mut vec![]).from_hex("012345");
```
here and elsewhere where you don't actually need to refer to `out` otherwise? (Don't feel strongly about this; this **is** clearer.)

Reply:
Unfortunately this is not possible. The anonymous temporary `vec![]` would life within the scope of the `Encoder::new` invocation only, i.e. dropped after the call, but it would be borrowed for the entire lifetime of `enc`. In other words `vec![]` would be dropped while still being borrowed by `enc`.

```
error[E0716]: temporary value dropped while borrowed
   --> neqo-common/src/codec.rs:459:37
    |
459 |         let enc = Encoder::new(&mut vec![]).from_hex("012345");
    |                                     ^^^^^^                    - temporary value is freed at the end of this statement
    |                                     |
    |                                     creates a temporary value which is freed while still in use
460 |         let mut dec = enc.as_decoder();
    |                       --- borrow later used here
    |
    = note: this error originates in the macro `vec` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider using a `let` binding to create a longer lived value
    |
459 ~         let binding = vec![];
460 ~         let enc = Encoder::new(&mut binding).from_hex("012345");
    |

For more information about this error, try `rustc --explain E0716`.
```